demo/README.md

```diff
@@ -0,0 +1,169 @@
+#Demo
+
+Prior to the release of the dynamic provisioning feature, [promoted to beta in Kubernetes 1.4](http://blog.kubernetes.io/2016/10/dynamic-provisioning-and-storage-in-kubernetes.html), administrators had to manually create new storage volumes and corresponding `PersistentVolumes`. Now, administrators can define `StorageClasses` to enable Kubernetes to create `PersistentVolumes` on-demand, and Kubernetes ships with many [provisioners](http://kubernetes.io/docs/user-guide/persistent-volumes/#provisioner) to specify in `StorageClasses` definitions.
+
+But what if Kubernetes does not yet include a provisioner for your preferred volume plugin? Kubernetes 1.5 introduces support for [external or out-of-tree provisioners](https://github.com/kubernetes/kubernetes/pull/30285): it now respects any `StorageClass` that specifies a provisioner it doesn't recognize, letting an external provisioner, which can be written, shipped and run independent of Kubernetes, handle it instead. [nfs-provisioner](https://github.com/kubernetes-incubator/nfs-provisioner) is an example of such an external provisioner. 
+
+nfs-provisioner creates NFS-backed PV's, leveraging the NFS volume plugin of Kubernetes, so given the ubiquity of NFS it will work almost anywhere. It's ideal for local clusters and dev work, any place a PV is wanted but not the manual work of creating one. We'll demonstrate how to get it quickly up and running, following a variation of the Kubernetes repo's [NFS example](https://github.com/kubernetes/kubernetes/tree/release-1.5/examples/volumes/nfs).
+
+Because it just needs to be able to talk to the Kubernetes API (using [client-go](https://github.com/kubernetes/client-go)), there is a lot of flexibility in how it is run. It can be run in Kubernetes as a `Pod`, `Deployment`, `StatefulSet`, or `DaemonSet`; as a standalone Docker container; or as a standalone binary. We'll demonstrate running as a deployment. All the yaml files we'll use can be found under the `demo` directory of the repo, [here](https://github.com/kubernetes-incubator/nfs-provisioner/tree/master/demo).
```

Comment:
I think this kind of freedom is counter intuitive. We should recommend one preferred way how to run the provisioner and somewhere below just mention that it can be run in many other ways, where we would describe pros and cons of every deployment method and with examples.